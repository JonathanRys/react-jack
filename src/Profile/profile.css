.Profile_main {
  background: #eee;
  border: 1px solid black;
  border-radius: 10px;
  min-width: 240px;
  margin: 15px;
  padding: 25px;
  padding-top: 20px;
}

.avatar {
  background: #fff;
  border: 1px solid black;
  border-radius: 5px;
  height: 52px;
  margin: 15px 5px;
  width: 52px;
}

.selected {
  box-shadow: inset 0 1px 2px rgba(27, 31, 35, 0.075),
    0 0 0 0.2em rgba(3, 102, 214, 0.3);
}
