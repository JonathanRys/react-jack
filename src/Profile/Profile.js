import React, { Component } from "react";
import PropTypes from "prop-types";

import { Button, Form, FormGroup, Label, Input } from "reactstrap";

import "./profile.css";

export default class Profile extends Component {
  constructor(props) {
    super(props);

    this.state = {
      playerName: props.name,
      playerAvatar: props.avatar
    };
  }

  onChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  onClick = () => {
    this.props.setName(this.state.playerName);
    this.props.setAvatar(this.state.playerAvatar);
  };

  render() {
    return (
      <Form className="Profile_main">
        <FormGroup>
          <Label for="playerName">Player name</Label>
          <Input
            type="text"
            name="playerName"
            id="playerName"
            onChange={this.onChange}
            value={this.state.playerName}
          />
        </FormGroup>
        <FormGroup>
          <Label for="avatar">Avatar</Label>
          <Input
            type="text"
            name="playerAvatar"
            id="playerAvatar"
            onChange={this.onChange}
            value={this.state.playerAvatar}
          />
          <img className="avatar" alt="avatar" src={this.props.avatar} />
        </FormGroup>
        <Button onClick={this.onClick}>Change</Button>
      </Form>
    );
  }
}

Profile.propTypes = {
  name: PropTypes.string.isRequired,
  avatar: PropTypes.string.isRequired,
  setName: PropTypes.func.isRequired,
  setAvatar: PropTypes.func.isRequired
};
