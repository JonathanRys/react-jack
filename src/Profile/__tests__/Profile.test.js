import React from "react";
import "../../setupTests";

import { Button } from "reactstrap";

import { shallow } from "enzyme";

import Profile from "../Profile";

const mockProps = {
  name: "Lester",
  avatar: "./images/default.png",
  setName: jest.fn(),
  setAvatar: jest.fn()
};

describe("Test Profile component", () => {
  it("sets new player name", () => {
    const component = shallow(<Profile {...mockProps} />);
    expect(component).toMatchSnapshot();
  });

  it("handles onChange on the name", () => {
    const component = shallow(<Profile {...mockProps} />);

    const event = {
      target: {
        name: "playerAvatar",
        value: "Lester"
      }
    };
    const input = component.find("#playerName");
    input.simulate("change", event);
    expect(component.state().playerName).toEqual("Lester");

    const button = component.find(Button);
    button.simulate("click");

    expect(mockProps.setName).toHaveBeenCalledWith("Lester");
  });

  it("handles onChange on the avatar", () => {
    const component = shallow(<Profile {...mockProps} />);

    const event = {
      target: {
        name: "playerAvatar",
        value: "./images/dealer.png"
      }
    };

    const input = component.find("#playerAvatar");
    input.simulate("change", event);
    expect(component.state().playerAvatar).toEqual("./images/dealer.png");

    const button = component.find(Button);
    button.simulate("click");
    expect(mockProps.setAvatar).toHaveBeenCalledWith("./images/dealer.png");
  });

  it("handles onClick", () => {});
});
