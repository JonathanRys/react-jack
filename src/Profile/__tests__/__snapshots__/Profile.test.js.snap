// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Profile component sets new player name 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile
    avatar="./images/default.png"
    changed={false}
    name="Lester"
    setAvatar={[MockFunction]}
    setName={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <FormGroup
          tag="div"
        >
          <Label
            for="playerName"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Player name
          </Label>
          <Input
            id="playerName"
            name="playerName"
            onChange={[Function]}
            type="text"
            value="Lester"
          />
        </FormGroup>,
        <FormGroup
          tag="div"
        >
          <Label
            for="avatar"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Avatar
          </Label>
          <br />
          <img
            alt="default avatar"
            className="avatar selected"
            id="defaultAvatar"
            onClick={[Function]}
            src="./images/default.png"
          />
          <img
            alt="male avatar"
            className="avatar "
            id="maleAvatar"
            onClick={[Function]}
            src="./images/male.png"
          />
          <img
            alt="female avatar"
            className="avatar "
            id="femaleAvatar"
            onClick={[Function]}
            src="./images/female.png"
          />
        </FormGroup>,
        <Button
          color="secondary"
          disabled={true}
          onClick={[Function]}
          tag="button"
        >
          Change
        </Button>,
      ],
      "className": "Profile_main",
      "tag": "form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Label
              for="playerName"
              tag="label"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              Player name
            </Label>,
            <Input
              id="playerName"
              name="playerName"
              onChange={[Function]}
              type="text"
              value="Lester"
            />,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Player name",
              "for": "playerName",
              "tag": "label",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": "Player name",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "playerName",
              "name": "playerName",
              "onChange": [Function],
              "type": "text",
              "value": "Lester",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Label
              for="avatar"
              tag="label"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              Avatar
            </Label>,
            <br />,
            <img
              alt="default avatar"
              className="avatar selected"
              id="defaultAvatar"
              onClick={[Function]}
              src="./images/default.png"
            />,
            <img
              alt="male avatar"
              className="avatar "
              id="maleAvatar"
              onClick={[Function]}
              src="./images/male.png"
            />,
            <img
              alt="female avatar"
              className="avatar "
              id="femaleAvatar"
              onClick={[Function]}
              src="./images/female.png"
            />,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Avatar",
              "for": "avatar",
              "tag": "label",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": "Avatar",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "default avatar",
              "className": "avatar selected",
              "id": "defaultAvatar",
              "onClick": [Function],
              "src": "./images/default.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "male avatar",
              "className": "avatar ",
              "id": "maleAvatar",
              "onClick": [Function],
              "src": "./images/male.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "female avatar",
              "className": "avatar ",
              "id": "femaleAvatar",
              "onClick": [Function],
              "src": "./images/female.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Change",
          "color": "secondary",
          "disabled": true,
          "onClick": [Function],
          "tag": "button",
        },
        "ref": null,
        "rendered": "Change",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <FormGroup
            tag="div"
          >
            <Label
              for="playerName"
              tag="label"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              Player name
            </Label>
            <Input
              id="playerName"
              name="playerName"
              onChange={[Function]}
              type="text"
              value="Lester"
            />
          </FormGroup>,
          <FormGroup
            tag="div"
          >
            <Label
              for="avatar"
              tag="label"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              Avatar
            </Label>
            <br />
            <img
              alt="default avatar"
              className="avatar selected"
              id="defaultAvatar"
              onClick={[Function]}
              src="./images/default.png"
            />
            <img
              alt="male avatar"
              className="avatar "
              id="maleAvatar"
              onClick={[Function]}
              src="./images/male.png"
            />
            <img
              alt="female avatar"
              className="avatar "
              id="femaleAvatar"
              onClick={[Function]}
              src="./images/female.png"
            />
          </FormGroup>,
          <Button
            color="secondary"
            disabled={true}
            onClick={[Function]}
            tag="button"
          >
            Change
          </Button>,
        ],
        "className": "Profile_main",
        "tag": "form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Label
                for="playerName"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                Player name
              </Label>,
              <Input
                id="playerName"
                name="playerName"
                onChange={[Function]}
                type="text"
                value="Lester"
              />,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Player name",
                "for": "playerName",
                "tag": "label",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": "Player name",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "playerName",
                "name": "playerName",
                "onChange": [Function],
                "type": "text",
                "value": "Lester",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Label
                for="avatar"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                Avatar
              </Label>,
              <br />,
              <img
                alt="default avatar"
                className="avatar selected"
                id="defaultAvatar"
                onClick={[Function]}
                src="./images/default.png"
              />,
              <img
                alt="male avatar"
                className="avatar "
                id="maleAvatar"
                onClick={[Function]}
                src="./images/male.png"
              />,
              <img
                alt="female avatar"
                className="avatar "
                id="femaleAvatar"
                onClick={[Function]}
                src="./images/female.png"
              />,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Avatar",
                "for": "avatar",
                "tag": "label",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": "Avatar",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "default avatar",
                "className": "avatar selected",
                "id": "defaultAvatar",
                "onClick": [Function],
                "src": "./images/default.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "male avatar",
                "className": "avatar ",
                "id": "maleAvatar",
                "onClick": [Function],
                "src": "./images/male.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "female avatar",
                "className": "avatar ",
                "id": "femaleAvatar",
                "onClick": [Function],
                "src": "./images/female.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Change",
            "color": "secondary",
            "disabled": true,
            "onClick": [Function],
            "tag": "button",
          },
          "ref": null,
          "rendered": "Change",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
